internal/server/server.go

```diff
@@ -0,0 +1,63 @@
+package server
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"tcp-load-balancer/internal/upstream"
+)
+
+// LoadBalancer is a TCP load balancer with methods for handling connections from clients to hosts.
+type LoadBalancer struct {
+	// listener is the TCP listener for this load balancer.
+	listener *net.TCPListener
+
+	// hosts is the list of upstream hosts
+	hosts []*upstream.TcpHost
```

Comment:
I think a RWMutex makes the most sense here, because we'll read from this slice much more often than hosts are added, and concurrent reads don't need to block each other. 

Adding this now.